framework/processmoduletest/supervisor_test.go

```
@@ -0,0 +1,535 @@
+	// serveProxy is invoked AFTER the route gate would have 404'd. With
+	// the gate bypassed (direct call), a not-Ready module returns 503.
+	if disabledRec.Code != http.StatusServiceUnavailable {
+		t.Errorf("disabled proxy: status = %d, want 503", disabledRec.Code)
+	}
+	if !hit503 {
+		t.Logf("note: did not observe the enable-time 503 window (spawn was fast)")
+	}
+}
+
+// waitForStateOn is waitForState parameterized by supervisor.
+func waitForStateOn(t *testing.T, sup *framework.ProcessModuleSupervisor, name string, want framework.ProcessState, timeout time.Duration) {
```

golangci-lint (2.12.2)

[error] func waitForStateOn is unused

(unused)